packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -150,6 +152,24 @@ export interface BasicCrawlerOptions<
      */
     handleRequestTimeoutSecs?: number;
 
+    /**
+     * A function to handle requests that failed less than `option.maxRequestRetries` times
+     * and would be retried by the crawler.
+     *
+     * The function receives the following object as the first argument:
+     * ```
+     * {
+     *     request: Request,
+     *     session: Session,
+     *     crawler: BasicCrawler,
+     * }
+     * ```
```

Comment:
same as the other comment, we can just link the interface (btw there can be more things right, e.g. proxy configuration/info)

in fact this should be all available already via the API docs thru the `ErrorHandler` type